holoviews/tests/plotting/bokeh/test_elementplot.py

```diff
@@ -993,3 +994,96 @@ def test_clim_percentile(self):
         low, high = plot.ranges[('Image',)]['z']['robust']
         assert low > 0
         assert high < 1
+
+class TestApplyHardBounds(TestBokehPlot):
+    def test_apply_hard_bounds(self):
+        """Test `apply_hard_bounds` with a single element."""
+        x_values = np.linspace(10, 50, 5)
+        y_values = np.array([10, 20, 30, 40, 50])
+        curve = Curve((x_values, y_values)).opts(apply_hard_bounds=True)
+        plot = bokeh_renderer.get_plot(curve)
+        self.assertEqual(plot.handles['x_range'].bounds, (10, 50))
```

Comment:
```suggestion
        assert plot.handles['x_range'].bounds == (10, 50)
```

Because I, at some point, want to transition from unit test to pytest. Same with other `self.assertEqual`.